pkg/causality/conflict_detector.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package causality
+
+import (
+	"sync"
+	"time"
+
+	"go.uber.org/atomic"
+
```

Comment:
Please delete this blank line.

Reply:
My IDE is separating internal and external imports. Do you think it would make the imports more readable?

Reply:
It looks OK. But it's different from what we were doing. And it may be difficult for people to follow this rule in practice.